File: scala/src/main/scala/tddmicroexercises/tirepressuremonitoringsystem/Alarm.scala
package tddmicroexercises.tirepressuremonitoringsystem

class Alarm(val sensor: Sensor) {

    val  LowPressureThreshold : Double = 17
    val HighPressureThreshold : Double = 21
    var alarmOn = false

    def check() {
        val psiPressureValue = sensor.popNextPressurePsiValue()
        alarmOn = psiPressureValue < LowPressureThreshold || HighPressureThreshold < psiPressureValue
    }

    def isAlarmOn() = alarmOn

}

File: Java/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Alarm.java
package tddmicroexercises.tirepressuremonitoringsystem;

public class Alarm {

    private final double LOW_PRESSURE_THRESHOLD = 17;
    private final double HIGH_PRESSURE_THRESHOLD = 21;

    private boolean alarmOn = false;
    private Sensor sensor;

    public Alarm(Sensor sensor) {
        this.sensor = sensor;
    }

    public Alarm() {
        this(new Sensor());
    }

    public void check() {
        double psiPressureValue = sensor.popNextPressurePsiValue();

        if (psiPressureValue < LOW_PRESSURE_THRESHOLD || HIGH_PRESSURE_THRESHOLD < psiPressureValue) {
            alarmOn = true;
        }
    }

    public boolean isAlarmOn() {
        return alarmOn;
    }
}
